src/coreComponents/mesh/generators/CellBlockManager.cpp

```diff
@@ -776,4 +776,333 @@ std::map< string, SortedArray< localIndex > > & CellBlockManager::getNodeSets()
   return m_nodeSets;
 }
 
+static array1d< real64 > gaussLobattoPoints( int order )
+{
+  array1d< real64 > GaussLobattoPts( order+1 );
+
+  switch( order )
+  {
+    case 1:
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = 1.0;
+      break;
+    case 2:
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = 0.0;
+      GaussLobattoPts[2] = 1.0;
+      break;
+    case 3:
+      static constexpr real64 sqrt5 = 2.2360679774997897;
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = -1./sqrt5;
+      GaussLobattoPts[2] = 1./sqrt5;
+      GaussLobattoPts[3] = 1.;
+      break;
+    case 4:
+      static constexpr real64 sqrt3_7 = 0.6546536707079771;
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = -sqrt3_7;
+      GaussLobattoPts[2] = 0.0;
+      GaussLobattoPts[3] = sqrt3_7;
+      GaussLobattoPts[4] = 1.0;
+      break;
+    case 5:
+      static constexpr real64 sqrt__7_plus_2sqrt7__ = 3.50592393273573196;
+      static constexpr real64 sqrt__7_mins_2sqrt7__ = 1.30709501485960033;
+      static constexpr real64 sqrt_inv21 = 0.218217890235992381;
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = -sqrt_inv21*sqrt__7_plus_2sqrt7__;
+      GaussLobattoPts[2] = -sqrt_inv21*sqrt__7_mins_2sqrt7__;
+      GaussLobattoPts[3] = sqrt_inv21*sqrt__7_mins_2sqrt7__;
+      GaussLobattoPts[4] = sqrt_inv21*sqrt__7_plus_2sqrt7__;
+      GaussLobattoPts[5] = 1.0;
+      break;
+  }
+  return GaussLobattoPts;
+}
+
+static void trilinearInterp( real64 const alpha,
+                             real64 const beta,
+                             real64 const gamma,
+                             real64 const (&X)[8][3],
+                             real64 (& coords)[3] )
+{
+  for( int i=0; i<3; i++ )
+  {
+    coords[i] = X[0][i]*( 1.0-alpha )*( 1.0-beta )*( 1.0-gamma )+
+                X[1][i]*    alpha    *( 1.0-beta )*( 1.0-gamma )+
+                X[2][i]*( 1.0-alpha )*    beta    *( 1.0-gamma )+
+                X[3][i]*    alpha    *    beta    *( 1.0-gamma )+
+                X[4][i]*( 1.0-alpha )*( 1.0-beta )*  gamma+
+                X[5][i]*    alpha    *( 1.0-beta )*  gamma+
+                X[6][i]*( 1.0-alpha )*    beta    *  gamma+
+                X[7][i]*    alpha    *    beta    *  gamma;
+  }
+}
+
+void CellBlockManager::generateHighOrderMaps( localIndex const order,
+                                              globalIndex const maxVertexGlobalID,
+                                              globalIndex const maxEdgeGlobalID,
+                                              globalIndex const maxFaceGlobalID,
```

Comment:
Speaking of those `max*GlobalId`, would it make sense to compute them directly in the in the `CellBlockManagerABC` or would it be dangerous/complicated?

Reply:
Good question Thomas, I was looking into this as well. I think it would not be dangerous, but would make the code more complicated (from more modifications and adding more functions for different objects point of views).

 As you can see, the "`maxGlobalIndex`" are set at `void ObjectManagerBase::setMaxGlobalIndex()`
And it was called from `mesh/ElementRegionManager.cpp` for `nodeManager.setMaxGlobalIndex()` and at `mesh/mpiCommunications/CommunicationTools.cpp` for face and edges `setMaxGlobalIndex`. 

So if we would like to have it in CellBlockManager.cpp. Firstly thing is to add three getMax*GlobalId functions at CellBlockManager.cpp after the communication stage, and also we need to change or added functions in NodeManager, FaceManager, and EdgeManager classes to pass the max*GlobalId to CellBlockManager. 

How do you think? Does it worth the effort to do this change or you think it would be simpler to pass the parameters as what we have now? I am looking forward to you suggestions.